src/rules/if_change_then_change.rs

```diff
@@ -99,47 +149,88 @@ pub fn ictc(
     let mut blocks: Vec<IctcBlock> = Vec::new();
 
     for block in all_blocks {
-        let block_lines = HashSet::from_iter(block.begin..block.end);
-        if !block_lines.is_disjoint(
-            modified_lines_by_path
-                .get(&block.path)
-                .unwrap_or(&HashSet::new()),
-        ) {
-            blocks.push(block);
+        match &block.thenchange {
+            ThenChange::MissingIf => {
+                blocks.push(block);
+            }
+            ThenChange::MissingThen => {
+                blocks.push(block);
+            }
```

Comment:
nit: can be combined:

```suggestion
            ThenChange::MissingIf | ThenChange::MissingThen => {
                blocks.push(block);
            }
```

Reply:
You are my rust tutor 🙇